File: Time Needed to Buy Tickets.py
class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        n = len(tickets)
        pass_k = False
        total = 0
        for i in range(n):
            if not pass_k:
                total += min(tickets[i], tickets[k])
            else:
                total += min(tickets[i], tickets[k] - 1)

            if i == k:
                pass_k = True
        return total

File: Time Needed to Buy Tickets.go
func timeRequiredToBuy(tickets []int, k int) int {
    n := len(tickets)
    pass_k := false
    total := 0
    for i := 0; i < n; i++ {
        if pass_k {
            total += min(tickets[i], tickets[k]-1)
        } else {
            total += min(tickets[i],tickets[k])
        }

        if i == k {
            pass_k = true
        }
    }
    return total
}
